### PR TITLE
Add FUN to clean_wqdata

### PR DESCRIPTION
**Pull request by sebdalgarno**
new FUN argument to clean_wqdata to allow user to summarize with 'mean', 'median' or 'max' of Value. Used to just get mean. Ability to get median and max was requested by Jessica Penno and Heather Thompson

### CODE REVIEW

**Comment by sebdalgarno** [comment]
sorry for the confusion in commits above - the first 'add FUN to calc_limits' is 1. misleading because it dealt with clean_wqdata and not calc_limits and 2. a first attempt.

The commit after the merge ('add FUN to clean_wqdata') is the commit finalizing the added functionality

File: R/clean-wqdata.R [hunk 1]
```diff
@@ -24,48 +24,48 @@ abs_dev <- function(x) {
...
+            signif(cv, 3), " for ", x$Variable[1], " on ", x$Date[1],
+            ".")
   }
   x[1, , drop = FALSE]
 }
 
-clean_wqdata_variable <- function(x, max_cv, messages) {
-  if (anyDuplicated(x$Date)) {
+clean_wqdata_variable <- function (x, max_cv, messages, FUN) {
+  if (anyDuplicated(x$Date))
```
Comment (ateucher) [inline R/clean-wqdata.R]: Nit: here and in the next block, I prefer curly braces around the expressions following `if` when there are multiple lines

Comment (sebdalgarno) [reply R/clean-wqdata.R]: I agree that passing a function is more elegant. However, I did this (along with argument name) to stay consistent with other functions in the package...e.g. `summarise_for_trends()` and `test_trends()`. I'm open to either, depending on whether you think consistency is particularly important

Comment (sebdalgarno) [reply R/clean-wqdata.R]: yes I will change so accepts a function instead of the name of it

File: R/clean-wqdata.R [hunk 2]
```diff
@@ -102,6 +95,7 @@ clean_wqdata_by <- function(x, max_cv, messages) {
 #' @param remove_blanks Should blanks be removed? Blanks are assumed to be denoted by
 #' a value of `"Blank..."` in the `SAMPLE_CLASS` column. Default `FALSE`
 #' @param messages A flag indicating whether to print messages.
+#' @param FUN The function to use for summaries, e.g. median, mean, or max.
```
Comment (ateucher) [inline R/clean-wqdata.R]: I like to be specific with the default value:

```suggestion
#' @param FUN The function to use for summaries, e.g. `median`, `mean`, or `max`. Default `mean`
```

Comment (sebdalgarno) [reply R/clean-wqdata.R]: sounds good. FYI as with the previous PR, the wording was copied from other functions in the package -  you could review ''test-trends.R' for other spots where FUN argument is used and described

Comment (ateucher) [reply R/clean-wqdata.R]: Thanks - A full review for consistency isn't in scope (at least for me), but we can improve iteratively as we come across things that are easy to fix!

Comment (sebdalgarno) [reply R/clean-wqdata.R]: OK that makes sense

**Pull request closed without merging**